src/components/PublicProfile.vue

```diff
@@ -42,17 +102,27 @@
 <script>
   import db from '../database';
   import util from '../util/util';
+  import firebase from 'firebase';
+  import AuthMixin from '../mixins/auth'
 
   const metricsRef = db.ref('metrics');
   const traitsRef = db.ref('traits');
   const schoolsRef = db.ref('schools');
 
   export default {
     name: 'profile',
+    mixins: [AuthMixin],
     data() {
       return {
+        traits: [],
```

Comment:
I would give these new 'data' members more descriptive names.
'traits' really refers to 'newReviewTraits' right?
You'd also want to do the same for the other metrics:
'cleanliness' -> 'newReviewCleanliness'
'loudness' -> 'newReviewLoudness' etc.